core/src/main/java/org/jruby/runtime/ivars/RawFieldVariableAccessor.java

```diff
@@ -0,0 +1,198 @@
+                if (realClass.getIsReifiedExtendedJavaClass() == Boolean.TRUE)
+                {
+                    // we are given a ConcreteJavaProxy, must unwrap first
+                    MethodHandle cjpUnwrap = Binder
+                            .from(realClass.getReifiedClass(), ConcreteJavaProxy.class, Object.class)
+                            .dropLast()
+                            .cast(Object.class, ConcreteJavaProxy.class)
+                            .invokeVirtual(RubyObjectSpecializer.LOOKUP, "unwrap");
+    
+                    MethodHandle wrapperSetConvert = 
+                            Binder
+                            .from(Object.class, realClass.getReifiedClass(), ConcreteJavaProxy.class, Object.class)
+                            .dropFirst(2)
+                            .cast(Object.class, IRubyObject.class)
+                            .insert(1, toJava)
+                            .invokeVirtual(RubyObjectSpecializer.LOOKUP, "toJava");
+                    
+                    MethodHandle wrapperSetter = 
+                            Binder
+                            .from(Object.class, Object.class, realClass.getReifiedClass(), ConcreteJavaProxy.class, Object.class)
+                            .permute(1,0)
+                            .cast(void.class, realClass.getReifiedClass(), basetype)
+                            .invoke(setter);
+    
+                    MethodHandle temp = MethodHandles.foldArguments(wrapperSetter, wrapperSetConvert);
+                    temp = MethodHandles.foldArguments(temp, cjpUnwrap);
+                    return temp;
+                }
+                else
+                {
+    
+                    MethodHandle wrapperSetConvert = 
+                            Binder
+                            .from(Object.class, realClass.getReifiedClass(), Object.class)
+                            .dropFirst()
+                            .cast(Object.class, IRubyObject.class)
+                            .insert(1, toJava)
+                            .invokeVirtual(RubyObjectSpecializer.LOOKUP, "toJava");
+                    
+                    MethodHandle wrapperSetter = 
+                            Binder
+                            .from(Object.class, Object.class, realClass.getReifiedClass(), Object.class)
+                            .permute(1,0)
+                            .cast(void.class, realClass.getReifiedClass(), basetype)
+                            .invoke(setter);
+    
+                    MethodHandle temp = MethodHandles.foldArguments(wrapperSetter, wrapperSetConvert);
+                    return temp;
+                }
+            }
+            
+        } catch (NoSuchMethodException |IllegalAccessException e) {
+            RaiseException r = Ruby.getGlobalRuntime().newRuntimeError("JRuby bug detected. Please file an issue if not already filed. Raw Field Accessor Setter gave errors: " + e.getMessage());
```

Comment:
could we rather use the runtime from `realClass` here, this doesn't seem right?

Reply:
Yes we should definitely use the realClass.getClassRuntime() rather than the global one.